js/ftx.js

```diff
@@ -478,30 +482,70 @@ module.exports = class ftx extends Exchange {
             const id = this.safeString (market, 'name');
             const baseId = this.safeString2 (market, 'baseCurrency', 'underlying');
             const quoteId = this.safeString (market, 'quoteCurrency', 'USD');
-            const type = this.safeString (market, 'type');
             const base = this.safeCurrencyCode (baseId);
             const quote = this.safeCurrencyCode (quoteId);
+            const isFuture = this.safeString (market, 'type') === 'future';
+            const contract = isFuture;
+            const name = this.safeString (market, 'name');
+            let symbol = base + '/' + quote;
+            const splitName = name.split ('-');
+            const settleId = contract ? 'USD' : undefined;
+            const spot = !contract;
+            const margin = !contract;
+            const swap = (this.safeString (splitName, 1) === 'PERP');
+            const option = false;
+            let futures = false;
+            let type = 'spot';
+            let expiry = undefined;
+            if (swap) {
+                type = 'swap';
+                symbol = base + '/' + quote + ':' + settleId;
+            } else if (contract) {
+                type = 'futures';
+                const expiryDates = this.safeValue (this.options, 'expiry');
+                if (id in expiryDates) {
+                    expiry = expiryDates[id];
+                } else {
+                    const arrayLength = splitName.length;
+                    const index = arrayLength - 1;
+                    expiry = this.safeValue (splitName, index);
+                }
+                symbol = base + '/' + quote + ':' + settleId + '-' + expiry;
+                futures = true;
+            }
             // check if a market is a spot or future market
-            const symbol = (type === 'future') ? this.safeString (market, 'name') : (base + '/' + quote);
-            const active = this.safeValue (market, 'enabled');
             const sizeIncrement = this.safeNumber (market, 'sizeIncrement');
             const priceIncrement = this.safeNumber (market, 'priceIncrement');
-            const precision = {
-                'amount': sizeIncrement,
-                'price': priceIncrement,
-            };
             result.push ({
                 'id': id,
                 'symbol': symbol,
                 'base': base,
                 'quote': quote,
                 'baseId': baseId,
                 'quoteId': quoteId,
+                'settleId': settleId,
                 'type': type,
-                'future': (type === 'future'),
-                'spot': (type === 'spot'),
-                'active': active,
-                'precision': precision,
+                'spot': spot,
+                'margin': margin,
+                'swap': swap,
+                'future': futures, // Deprecated, please use 'futures'
```

Comment:
I also have this, so that `future` will still work

Reply:
no it won't lets restore this aspect of the api so `type` is future pls